cranelift/codegen/meta/src/shared/instructions.rs

```diff
@@ -141,7 +141,8 @@ fn define_control_flow(
             Operand::new("c", ScalarTruthy).with_doc("Controlling value to test"),
             Operand::new("code", &imm.trapcode),
         ])
-        .can_trap(),
+        .can_trap()
+        .side_effects_idempotent(),
```

Comment:
Can we add a comment here describing the rationale a bit more (for our future selves as much as others)?

Something like: "`side_effects_idempotent` indicates that when we have more than one copy of this instruction, it is safe to merge (GVN) them. Traps have this characteristic: as long as we don't remove them, if we have already passed through (computed) one trap instruction on a given condition, it is safe to consider other traps on the same condition as redundant. In other words, a trap on value X is idempotent: once we observe its side effect (which will either terminate execution or do nothing), we don't need to observe it again."